### ISSUE TITLE
incompatible with `no-unused-vars` rule

### ISSUE THREAD

**Issue by CNSeniorious000** [sveltejs/eslint-plugin-svelte#971]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.17.0

### What version of `eslint-plugin-svelte` are you using?

2.46.1

### What did you do?

<details>
<summary>Configuration</summary>

```ts
import antfu from '@antfu/eslint-config'

export default antfu({
  svelte: true,
  typescript: true,
  formatters: true,
})
```
</details>

```svelte
{#if true}
  {#each [1, 2, 3] as a}
    {a}

    <!-- eslint-disable-next-line no-unused-vars -->
    {@const _ = console.error({ a })}

  {/each}
{/if}
```

### What did you expect to happen?

no error

### What actually happened?

```
ops! Something went wrong! :(

ESLint: 9.17.0

TypeError: Cannot read properties of undefined (reading 'length')
    at handleFixes (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/rules/no-unused-vars.js#cjs:1162:48)
    at Object.fix (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/rules/no-unused-vars.js#cjs:1471:48)
    at normalizeFixes (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/report-translator.js#cjs:193:28)
    at eval (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/report-translator.js#cjs:223:22)
    at Array.map (<anonymous>)
    at mapSuggestions (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/report-translator.js#cjs:217:10)
    at eval (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/report-translator.js#cjs:372:55)
    at FileContext.report (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/linter.js#cjs:1062:41)
    at Program:exit (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/rules/no-unused-vars.js#cjs:1458:33)
    at ruleErrorHandler (file:///home/projects/vitejs-vite-geepgmmf/node_modules/eslint/lib/linter/linter.js#cjs:1098:48)
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/CNSeniorious000/eslint-no-unused-vars-repro

### Additional comments

- Redirected from https://github.com/eslint/eslint/issues/19273#issuecomment-2564270111

**Comment by baseballyama** [comment]
To fix this, I think we need to implement our own extended rule.

**Comment by baseballyama** [comment]
I believe [[this PR](https://github.com/sveltejs/svelte-eslint-parser/pull/641)](https://github.com/sveltejs/svelte-eslint-parser/pull/641) fixes the issue.
I have confirmed that 3.0.0-next.18 resolves the problem.
